yarn-project/bb-prover/src/avm_proving.test.ts

```diff
@@ -179,7 +179,16 @@ describe('AVM WitGen, proof generation and verification', () => {
  * Helpers
  ************************************************************************/
 
-const proveAndVerifyAvmTestContract = async (functionName: string, calldata: Fr[] = [], mutatedCalldata: Fr[] = []) => {
+/**
+ * If assertionErrString is set, we expect a (non exceptional halting) revert due to a failing assertion and
+ * we check that the revert reason error contains this string. However, the circuit must correctly prove the
+ * execution.
+ */
+const proveAndVerifyAvmTestContract = async (
+  functionName: string,
+  calldata: Fr[] = [],
+  assertionErrString?: string,
+) => {
   const startSideEffectCounter = 0;
   const globals = GlobalVariables.empty();
   globals.timestamp = TIMESTAMP;
```

Comment:
Do we need this? It made sense when you were testing mutated data, but for the current test, you can just call a contract function that does `assert(false, "some message");` (maybe there is one already that just asserts?) and you'd be testing the same without having to mix timestamps in the middle.

Reply:
While a mismatched timestamp leads to sthg equivalent to assert(false, ...), I think it is also valuable to test this scenario. Relying on the positive test case to assume that timestamp goes through the assertion does not seem as robust to me. Or do you think we are more testing Noir than the AVM by doing so?